listenbrainz/webserver/views/user_timeline_event_api.py

```diff
@@ -238,6 +239,68 @@ def user_feed(user_name: str):
         'events': [event.dict() for event in all_events],
     }})
 
+@user_timeline_event_api_bp.route("/user/<user_name>/feed/events/delete", methods=['OPTIONS', 'POST'])
+@crossdomain(headers="Authorization, Content-Type")
+@ratelimit()
+def delete_feed_events(user_name):
+    '''
+    Delete those events from user's feed that they created themselves.
+    Supports deletion of recommendation, notification and pin events.
+    Along with the authorization token, post one of the following, according
+    to your need.
+    {
+        "event_type": "recording_recommendation",
+        "metadata": {
+            id: int
+        }
+    }
+    {
+        "event_type": "notification",
+        "metadata": {
+            id: int
+        }
+    }
+    {
+        "event_type": "recording_pin",
```

Comment:
I think it makes more sense to just accept an `id` here. Even the event type is not needed becuse we do not use it in the where clause while deleting.